### PR TITLE
Fix get_OS_version for MacOS

### PR DESCRIPTION
**Pull request by pdbain-ibm**
Also define platform-specific constants to reduce the complexity of the ifdefs
and reduce duplicated code.

Signed-off-by: Peter Bain <peter_bain@ca.ibm.com>

### CODE REVIEW

**Comment by charliegracie** [comment]
Can you add some comments to the code explaining why OS X is not using the same code as other `unix` style platforms? Thanks

**Comment by pdbain-ibm** [comment]
@charliegracie 
Thanks for the suggestion.  Done.

File: port/unix/omrsysinfo.c [hunk 1]
```diff
@@ -581,46 +584,66 @@ omrsysinfo_get_OS_type(struct OMRPortLibrary *portLibrary)
...
 	if (NULL == PPG_si_osVersion) {
 		int rc;
+#if !defined (OSX)
 		struct utsname sysinfo;
+#endif
 
 #ifdef J9ZOS390
 		rc = __osname(&sysinfo);
+#elif defined (OSX)
+			/* uname() on MacOS returns the version of the Darwin kernel.
```
Comment (charliegracie) [inline port/unix/omrsysinfo.c]: It looks like there is an extra tab for the comment. Could you align the comment with the code?  Thanks

Thanks for the comment as it is exactly what I was looking for!

**Comment by charliegracie** [comment]
@genie-omr build all

**Pull request merged**